lib/cc/analyzer/container.rb

```diff
@@ -0,0 +1,98 @@
+require "posix/spawn"
+
+module CC
+  module Analyzer
+    class Container
+      DEFAULT_TIMEOUT = 15 * 60 # 15m
+
+      def initialize(
+        image:,
+        name:,
+        command: nil,
+        log: NullContainerLog.new,
+        timeout: DEFAULT_TIMEOUT
+      )
+        @image = image
```

Comment:
> I haven't seen much of using instance variables directly

I'd have to do some kind of audit to be sure, but at least in this PR `Engine` also accesses its instance variables directly.

I was going to switch these over, but it's actually kind of annoying and error prone so I'm going to leave them as is (if that's OK with you). If you want to push for the `attr_reader` approach can you open a PR on the style guide?